packages/extension-browser/tests/background-script.ts

```diff
@@ -0,0 +1,443 @@
+import test from 'ava';
+import * as proxyquire from 'proxyquire';
+import { createSandbox, SinonSandbox } from 'sinon';
+
+import { FetchEnd, FetchStart } from 'hint/dist/src/lib/types/events';
+
+import { Config, Details, Results } from '../src/shared/types';
+
+import { awaitListener, stubEvent, stubGlobals, Globals } from './helpers/globals';
+
+const backgroundScriptPath = '../src/background-script';
+
+/**
+ * Require the background script using the provided `browser` APIs.
+ */
+const loadBackgroundScript = (globals: Globals) => {
+    proxyquire(backgroundScriptPath, {
+        './shared/globals': {
+            '@noCallThru': true,
+            ...globals
+        }
+    });
+    require(backgroundScriptPath);
+};
+
+/**
+ * Returns a method which can be invoked to trigger content script injection.
+ * Returned method accepts an argument specifying the `tabId` to inject to.
+ * Simulates receiving `webRequest.onResponseStarted` for the main frame of the tab.
+ *
+ * Note: Must be called to get the method *before* loading the background script.
+ *
+ * Note: Returned method must be called *after* loading the background script.
+ */
+const prepareContentScriptInjection = (sandbox: SinonSandbox, browser: typeof chrome) => {
+    const onMessagePromise = awaitListener(sandbox, browser.runtime.onMessage);
+    const onResponseStartedPromise = awaitListener(sandbox, browser.webRequest.onResponseStarted);
+
+    return async (tabId: number): Promise<void> => {
+
+        const onMessage = await onMessagePromise;
+
+        // Simulate receiving `Events.enable` from the devtools panel.
+        onMessage({ enable: {} }, { tab: { id: tabId } } as any, () => {});
+
+        const onResponseStarted = await onResponseStartedPromise;
+
+        // Simulate receiving `webRequest.onResponseStarted` for the main frame.
+        onResponseStarted({
+            frameId: 0,
+            fromCache: false,
+            method: 'GET',
+            parentFrameId: -1,
+            requestId: '1',
+            statusCode: 200,
+            statusLine: '',
+            tabId,
+            timeStamp: Date.now(),
+            type: 'main_frame',
+            url: '/'
+        });
+    };
+};
+
+test('It registers for messages.', async (t) => {
+    const sandbox = createSandbox();
+    const globals = stubGlobals();
+    const { browser } = globals;
+
+    const onMessagePromise = awaitListener(sandbox, browser.runtime.onMessage);
+
+    loadBackgroundScript(globals);
+
+    const onMessage = await onMessagePromise;
+
+    t.true(typeof onMessage === 'function', 'Background script provided a listener');
+
+    sandbox.restore();
+});
+
+test('It reloads the target when enabled.', async (t) => {
+    const sandbox = createSandbox();
+    const globals = stubGlobals();
+    const { browser } = globals;
+    const tabId = 5;
+
+    const reloadSpy = sandbox.stub(browser.tabs, 'reload');
```

Comment:
If you don't need to change the return value or anything, you can use a `spy` instead of an `stub`

Reply:
Heh, the variable name even suggests that was my intent. I'll fix it.